plotly/tools.py

```diff
@@ -1503,35 +1503,53 @@ def _unconvert_from_RGB_255(colors):
             return un_rgb_colors
 
     @staticmethod
-    def _map_array2color(array, colormap, vmin, vmax):
+    def _map_face2color(face, colormap, vmin, vmax):
         """
-        Normalize values in array by vmin/vmax and return plotly color strings.
+        Normalize facecolor values by vmin/vmax and return rgb-color strings
 
-        This function takes an array of values along with a colormap and a
-        minimum (vmin) and maximum (vmax) range of possible z values for the
-        given parametrized surface. It returns an rgb color based on the
-        relative position of zval between vmin and vmax
+        This function takes a tuple color along with a colormap and a minimum
+        (vmin) and maximum (vmax) range of possible mean distances for the
+        given parametrized surface. It returns an rgb color based on the mean
+        distance between vmin and vmax
 
         """
         if vmin >= vmax:
             raise exceptions.PlotlyError("Incorrect relation between vmin "
                                          "and vmax. The vmin value cannot be "
                                          "bigger than or equal to the value "
                                          "of vmax.")
-        # find distance t of zval from vmin to vmax where the distance
-        # is normalized to be between 0 and 1
-        t = (array - vmin) / float((vmax - vmin))
-        t_colors = FigureFactory._find_intermediate_color(colormap[0],
-                                                          colormap[1],
-                                                          t)
-        t_colors = t_colors * 255.
-        labelled_colors = ['rgb(%s, %s, %s)' % (i, j, k)
-                           for i, j, k in t_colors.T]
-        return labelled_colors
+        # find the normalized distance t of a triangle face between vmin and
+        #vmax where the distance is normalized between 0 and 1
+        t = (face - vmin) / float((vmax - vmin))
+
+        if len(colormap) <= 1:
+            t_color = colormap[0]
+            color = FigureFactory._convert_to_RGB_255(t_color)
+            color = FigureFactory._label_rgb(color)
+        else:
+            # account for colormaps with more than one color
+            incr = 1./(len(colormap) - 1)
+            low_color_index = int(t/incr)
+
+            if t == 1:
+                t_color = colormap[low_color_index]
```

Comment:
so, `colormap[np.exp(np.pi * np.sqrt(-1))]`? ;)

On Wed, Jun 1, 2016 at 11:19 AM, Andrew <notifications@github.com> wrote:

> In plotly/tools.py
> <https://github.com/plotly/plotly.py/pull/493#discussion_r65414977>:
>
> > -        return labelled_colors
> > +        # find the normalized distance t of a triangle face between vmin and
> > +        #vmax where the distance is normalized between 0 and 1
> > +        t = (face - vmin) / float((vmax - vmin))
> > +
> > +        if len(colormap) <= 1:
> > +            t_color = colormap[0]
> > +            color = FigureFactory._convert_to_RGB_255(t_color)
> > +            color = FigureFactory._label_rgb(color)
> > +        else:
> > +            # account for colormaps with more than one color
> > +            incr = 1./(len(colormap) - 1)
> > +            low_color_index = int(t/incr)
> > +
> > +            if t == 1:
> > +                t_color = colormap[low_color_index]
>
> Similar to https://github.com/plotly/plotly.py/pull/493/files#r65412272.
> Why not just do colormap[-1] here? It's much easier to understand. You're
> just doing colormap[1 / (1 / (len(colormap) - 1))].
>
> [image: image]
> <https://camo.githubusercontent.com/4815359abf15ffcb436e4b27c7bdcda27bbc93da/687474703a2f2f696d67732e786b63642e636f6d2f636f6d6963732f655f746f5f7468655f70695f74696d65735f692e706e67>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/plotly/plotly.py/pull/493/files/c5339c1b9e8c3b896c6a70f1c3241630df67aba9#r65414977>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABwSHaS4pYXxTQM_gb0rE1D9z0zvh-yRks5qHc08gaJpZM4Ip-KO>
> .
>